cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -154,6 +166,13 @@ public static HiveSupport checkHiveSupport(@Nullable ClassLoader hiveClassLoader
       }
     }
 
+    boolean useLatestVersionForUnknown = cConf.get(Constants.Explore.HIVE_AUTO_LATEST_VERSION).equals(
+      cConf.get(Constants.Explore.HIVE_VERSION_FOR_UNKNOWN_VERSION));
+    if (useLatestVersionForUnknown && CDH_PATTERN.matcher(hadoopVersion).matches()) {
+      LOG.info("CDH Hadoop version '{}' is unknown. Using the latest Hive version available in CDAP. ", hadoopVersion);
```

Comment:
Yes - say "Hive version {} is unsupported". Look at the comment below to see how to get the Hive version.